server/http/helloapp/auth.go

```diff
@@ -70,25 +70,24 @@ func (h *helloapp) startOAuth2Connect(userID string, callOnComplete apps.Call) (
 }
 
 func (h *helloapp) finishOAuth2Connect(userID string, token oauth2.Token, payload []byte) {
-	call := apps.Call{}
-	err := json.Unmarshal(payload, &call)
+	call, err := apps.UnmarshalCall(payload)
 	if err != nil {
 		return
 	}
-	call.Request.Context.AppID = AppID
+	call.Context.AppID = AppID
 
 	// TODO 2/5 we should wrap the OAuther for the users as a "service" so that
 	//  - startOAuth2Connect is a Call
 	//  - payload for finish should be a Call
-	//  - a Wish can check the presence of the acting user's OAuth2 token, and
+	//  - a Call can check the presence of the acting user's OAuth2 token, and
 	//    return Call startOAuth2Connect(itself)
 	// for now hacking access to apps object and issuing the call from within
 	// the app.
 
-	cr, _ := h.apps.Client.PostWish(&call)
+	cr, _ := h.apps.Client.PostCall(call)
```

Comment:
I know this is no new code, but wondering now. Why the helloapp cast a call when it finishes the OAuth connect? Wouldn't it simpler to just "do something", instead of going through the process to perform an http call to itself?

Reply:
@larkox and I discussed offline how it should be put in the OAuther, as a `Call`